### PR TITLE
Update contributing documents

### PR DESCRIPTION
**Pull request by kmova**
Signed-off-by: kmova <kiran.mova@openebs.io>

Fixes https://github.com/openebs/openebs/issues/459

**What this PR does / why we need it**: Update the openebs/maya/README.md to reflect the latest architecture changes and also include a diagram for better readability. Provide links to the correct contributing guides. 

Similarly updated the CONTRIBUTING.md with some help text for new contributors. 
Restructured the developer.md into different sections that have to be followed by contributors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,34 +3,51 @@
...
+- **Control Plane Components** - These are containers that are initialized as part of enabling the OpenEBS in a Kubernetes cluster.
+  * *maya-apiserver* helps with creation of CAS Volumes and provides API endpoints to manage those volumes. *maya-apiserver* can also be considered as an template engine that can be easily extended to support any kind of CAS storage solutions. It takes as input a set of CAS Templates that are converted into CAS K8s YAMLs based on the user requests.
+  * *provisioner* is an implementation of Kubernetes Dynamic Provisioner that processes the PVC requests by interacting with maya-apiserver. 
+
+- **CAS Side-car Components** - These are adapter components that help with managing the CAS containers that don't inherently come up the required endpoints. For example: 
+  - *maya-exporter* helps in providing a metrics endpoint to the CAS container.
```
Comment (utkarshmani1997) [inline README.md]: maya-volume-exporter

Comment (kmova) [reply README.md]: good point @utkarshmani1997  As we are making the maya-volume-exporter to include more objects to monitor, it makes sense to rename maya-volume-exporter to exporter. Will be tracked under : https://github.com/openebs/openebs/issues/1379

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,46 +1,56 @@
-# Contributing to OpenEBS / Maya
+# Contributing to OpenEBS Maya
 
-Are you ready to enhance Maya. Great!!!
+OpenEBS uses the standard GitHub pull requests process to review and accept contributions.  There are several areas that could use your help. For starters, you could help with improving the sections is this document by either creating a new issue describing the improvement or submitting a pull request to this repository. The issues for the various openebs components (including maya components) are maintained in [openebs/openebs](https://github.com/openebs/openebs/issues) repository.
```
Comment (mahebbar) [inline CONTRIBUTING.md]: For starters, = To begin with, you could help in improving...
sections is this = sections in this

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,46 +1,56 @@
...
 
-Feel free to contribute by:
-- raising issues
-- participate in commenting/voting on the issues
-- create new PRs for Documentation or Code
+* If you are a first time contributor, please see the below [Steps to Contribute](#steps-to-contribute)
```
Comment (mahebbar) [inline CONTRIBUTING.md]: see the below = see Steps to Contribute.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,46 +1,56 @@
...
+* If you have documentation improvement ideas, go ahead and create a pull request. See [Pull Request checklist](#pull-request-checklist)
+* If you would like to make code contributions, please start with [Setting up the Development Environment](#setting-up-your-development-environment).
+* If you would like to work on something more involved, please connect with the OpenEBS Contributors. See [OpenEBS Community](https://github.com/openebs/openebs/tree/master/community)
 
-## Contributing to Source Code
+## Steps to Contribute
 
+OpenEBS is a Apache 2.0 Licensed project and all your commits should be signed with Developer Certificate of Origin. See [Sign your work](#sign-your-work).
```
Comment (mahebbar) [inline CONTRIBUTING.md]: is an Apache...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,46 +1,56 @@
...
 
-## Contributing to Source Code
+## Steps to Contribute
 
+OpenEBS is a Apache 2.0 Licensed project and all your commits should be signed with Developer Certificate of Origin. See [Sign your work](#sign-your-work). 
 
-### Setting up your Development Environment
+* Find an issue to work on or create a new issue. The issues are maintained at [openebs/openebs](https://github.com/openebs/openebs/issues). You can pick up from a list of [good-first-issues](https://github.com/openebs/openebs/labels/good%20first%20issue).
+* Claim your issue by commenting your intent to work on it to avoid duplication of efforts. 
+* Fork the repository on GitHub
```
Comment (mahebbar) [inline CONTRIBUTING.md]: Github.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,46 +1,56 @@
...
-vagrant up master-01
-vagrant ssh master-01
-#Only for the first time
-make bootstrap
-make deps
-make dev
-```
+## Pull Request Checklist
+* Rebase to the current master branch before submitting your pull request.
+* Commits should be as small as possible. Each commit should follow these checklist:
```
Comment (mahebbar) [inline CONTRIBUTING.md]: follow these = follow the checklist below:

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,46 +1,56 @@
...
+  - For code-changes, add tests relevant to the fixed bug or new feature.
```
Comment (mahebbar) [inline CONTRIBUTING.md]: code-changes = code changes,

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,46 +1,56 @@
...
-#Only for the first time
-make bootstrap
-make deps
-make dev
-```
+## Pull Request Checklist
+* Rebase to the current master branch before submitting your pull request.
+* Commits should be as small as possible. Each commit should follow these checklist:
+  - For code-changes, add tests relevant to the fixed bug or new feature. 
+  - Pass the compile and tests - includes spell checks, formatting.
```
Comment (mahebbar) [inline CONTRIBUTING.md]: spell checks and formatting.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,46 +1,56 @@
...
+  - Commit header (first line) should convey what changed
+  - Commit body should include the details like why the changes is required and how the change has been made
```
Comment (mahebbar) [inline CONTRIBUTING.md]: include details such as why the changes are required...

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,46 +1,56 @@
...
 
-### Sign your work
-
-We use the Developer Certificate of Origin (DCO) as a additional safeguard
-for the OpenEBS project. This is a well established and widely used
-mechanism to assure contributors have confirmed their right to license
-their contribution under the project's license.
-Please read [dcofile](https://github.com/openebs/openebs/blob/master/contribute/developer-certificate-of-origin).
-If you can certify it, then just add a line to every git commit message:
+We use the Developer Certificate of Origin (DCO) as a additional safeguard for the OpenEBS project. This is a well established and widely used mechanism to assure contributors have confirmed their right to license their contribution under the project's license. Please read [dcofile](https://github.com/openebs/openebs/blob/master/contribute/developer-certificate-of-origin). If you can certify it, then just add a line to every git commit message:
```
Comment (mahebbar) [inline CONTRIBUTING.md]: as an additional security (instead of safeguard?)
to assure contributors = to assure that contributors...

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,46 +1,56 @@
...
   Signed-off-by: Random J Developer <random@developer.example.org>
 ````
 
-Use your real name (sorry, no pseudonyms or anonymous contributions).
-If you set your `user.name` and `user.email` git configs, you can sign your
-commit automatically with `git commit -s`. You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases)
-like `git config --global alias.ci 'commit -s'`. Now you can commit with
-`git ci` and the commit will be signed.
+Use your real name (sorry, no pseudonyms or anonymous contributions). The email id should match the email id provided in your GitHub profile. 
+If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`.
```
Comment (mahebbar) [inline CONTRIBUTING.md]: git configs = git configuration

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,46 +1,56 @@
...
+If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`. 
+
+You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases) like `git config --global alias.ci 'commit -s'`. Now you can commit with `git ci` and the commit will be signed.
+
+## Setting up your Development Environment
+
+This project is implemented using Go and uses the standard golang tools for development and build. In addition, this project heavily relies on Docker and Kubernetes. It is expected that the contributors:
+- are familiar with working with Go
+- are familiar with Docker containers
+- are familiar with Kubernetes and have access to a Kubernetes Cluster or Minikube to test the changes.
```
Comment (mahebbar) [inline CONTRIBUTING.md]: Kubernetes cluster

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -1,46 +1,56 @@
...
+
+For setting up Development environment on your local host, see the detailed instruction [here](./docs/developer.md).
```
Comment (mahebbar) [inline CONTRIBUTING.md]: instructions

File: README.md [hunk 14]
```diff
@@ -3,34 +3,51 @@
...
 
-*This repository hosts the source code for OpenEBS Storage Orchestration Engine.* 
+*Visit https://docs.openebs.io to learn about Container Attached Storage(CAS) and full documentation on using OpenEBS Maya*
 
-OpenEBS Storage Orchestration (aka *Maya*, meaning *Magic*) aims at managing storage for millions of containers with deceptive simplicity. Maya is the control plane for OpenEBS Storage. Maya is developed with a focused *goal* of easing(and at times completely eliminating) the Storage Ops for containers like 
-- managing Storage Volumes (Persistent Volumes) and 
-- managing Storage Infrastructure (Physical Storage - Local/Cloud Disks/SSDs/Cache, Controllers, Networks). 
+OpenEBS Maya extends the capabilities of Kubernetes to orchestrate CAS (aka Container Native) Storage Solutions like OpenEBS Jiva, OpenEBS cStor, etc. *Maya* (meaning *Magic*), seamlessly integrates into the Kubernetes Storage Workflow and helps provision and manages the CAS based Storage Volumes. The core-features of *Maya* include:
```
Comment (mahebbar) [inline README.md]: helps provision and manage ....

File: README.md [hunk 15]
```diff
@@ -3,34 +3,51 @@
...
+- Maintaining the inventory of the underlying disks on the Kubernetes Nodes 
+- Managing the allocation of Disks to CAS Storage Engines
+- Provisioning of CAS Volumes by interfacing with K8s Dynamic Volume Provisioner
+- Managing the high availability of the CAS volumes by tuning the scheduling parameters of CAS Deployments (Pods)
+- Provide adapters to CAS Volumes to interact with Kubernetes and related infrastructure components like Prometheus, Grafana etc.
 
-OpenEBS Maya allows you to manage storage across multiple zones (aka clusters/environments), that are co-located or geographically separated and can also run from within a single host. Maya can move the storage across different tiers based on the application needs (volume migration). Maya can learn and adapt to the changing environment through machine learning and data analytics. 
+*Maya* orchestration and management capabilities are delivered through a set of services and tools. Currently these services support deploying the CAS Storage Solutions in Kubernetes Clusters. In the future, these can be extended to support other Container Orchestrators.
```
Comment (mahebbar) [inline README.md]: In the future = In future

File: README.md [hunk 16]
```diff
@@ -3,34 +3,51 @@
...
-## Design
-
-Maya - *Storage Orchestration* functionality is delivered through a set of services and tools that seamlessly integrate *Maya* into your Container Orchestrators like Kubernetes, Docker Swarm, etc.,  Maya comprises of several components, which are themself delivered as container images. 
+## Maya Architecture 
+![](./docs/openebs-maya-architecture.png)
 
 **Maya** components can be broadly classified based on their deployment type as follows:
 
-- **Cluster Components** - like API Server (maya-apiserver) that helps in processing the requests for creating new OpenEBS Volumes. The API server, will use the container orchestrators to provision the OpenEBS Volume Containers. Another example is the Smart Analytics (maya-mulebot) engine that gathers the data via the machine learning probes and runs heuristics analysis to optimize storage deployment. 
+- **Control Plane Components** - These are containers that are initialized as part of enabling the OpenEBS in a Kubernetes cluster.
```
Comment (mahebbar) [inline README.md]: enabling the OpenEBS = enabling OpenEBS

File: README.md [hunk 17]
```diff
@@ -3,34 +3,51 @@
...
+  * *maya-apiserver* helps with creation of CAS Volumes and provides API endpoints to manage those volumes. *maya-apiserver* can also be considered as an template engine that can be easily extended to support any kind of CAS storage solutions. It takes as input a set of CAS Templates that are converted into CAS K8s YAMLs based on the user requests.
```
Comment (mahebbar) [inline README.md]: as an template = as a template
CAS Templates = CAS templates
on the user = on user

File: README.md [hunk 18]
```diff
@@ -3,34 +3,51 @@
...
+  * *provisioner* is an implementation of Kubernetes Dynamic Provisioner that processes the PVC requests by interacting with maya-apiserver. 
+
+- **CAS Side-car Components** - These are adapter components that help with managing the CAS containers that don't inherently come up the required endpoints. For example:
```
Comment (mahebbar) [inline README.md]: don't = do not 
come up with the

File: README.md [hunk 19]
```diff
@@ -3,34 +3,51 @@
...
+- **Control Plane Components** - These are containers that are initialized as part of enabling the OpenEBS in a Kubernetes cluster.
+  * *maya-apiserver* helps with creation of CAS Volumes and provides API endpoints to manage those volumes. *maya-apiserver* can also be considered as an template engine that can be easily extended to support any kind of CAS storage solutions. It takes as input a set of CAS Templates that are converted into CAS K8s YAMLs based on the user requests.
+  * *provisioner* is an implementation of Kubernetes Dynamic Provisioner that processes the PVC requests by interacting with maya-apiserver. 
+
+- **CAS Side-car Components** - These are adapter components that help with managing the CAS containers that don't inherently come up the required endpoints. For example: 
+  - *maya-exporter* helps in providing a metrics endpoint to the CAS container.
+  - *cas-mgmt* components can be attached as side-cars for helping to store/retrieve configuration information from Kubernetes Config Store(etcd). For cStor CAS solution, cstor-pool-mgmt is one such *cas-mgmt* component.
```
Comment (mahebbar) [inline README.md]: space (etcd).

File: README.md [hunk 20]
```diff
@@ -3,34 +3,51 @@
...
 
-## Start developing Maya
+## Community
 
-Head over to the [developer's documentation](https://github.com/openebs/maya/blob/master/docs/developer.md) for more details.
+See the [OpenEBS Community page](https://github.com/openebs/openebs/tree/master/community) for reaching out to the OpenEBS Developers.
 
-### Dependency management
-Maya uses [golang dep] to manage dependencies. Usage can be found on [dep README].
+## More Info
```
Comment (mahebbar) [inline README.md]: More Information

File: README.md [hunk 21]
```diff
@@ -3,34 +3,51 @@
...
+- Design proposals for *Maya* components are located at [OpenEBS Designs directory](https://github.com/openebs/openebs/tree/master/contribute/design)
+- The issues related to *Maya* are logged under [openebs/openebs](https://github.com/openebs/openebs/issues).
+- To build Maya from the source code see [developer's documentation].
```
Comment (mahebbar) [inline README.md]: code, see

File: docs/developer.md [hunk 22]
```diff
@@ -1,15 +1,18 @@
-# Contributing Guide
+# Development Workflow
 
-To contribute openebs/maya firstly, you have to fork the openebs/maya repository. But before that you need to create an issue or pick an existing issue from [open issues](https://github.com/openebs/openebs/issues)
+## Prerequisites
 
-In any case, before you start working on your issue, sync your repository with the upstream openebs/maya master. Syncing ensures your repository has the latest changes.
+* You have Go 1.9 installed on your host/development machine
```
Comment (mahebbar) [inline docs/developer.md]: machine.

File: docs/developer.md [hunk 23]
```diff
@@ -1,15 +1,18 @@
...
+* You have Docker installed on your host/development machine. Docker is required for building maya components container images and to push them into Kubernetes Cluster for testing.
```
Comment (mahebbar) [inline docs/developer.md]: maya component

File: docs/developer.md [hunk 24]
```diff
@@ -1,15 +1,18 @@
...
+
+### Fork in the cloud
 
 1. Visit https://github.com/openebs/maya
 2. Click `Fork` button (top right) to establish a cloud-based fork.
 
-### 2. Clone fork to local storage
+### Clone fork to local machine
 
 Place openebs/maya' code on your `GOPATH` using the following cloning procedure.
```
Comment (mahebbar) [inline docs/developer.md]: following cloning command.

File: docs/developer.md [hunk 25]
```diff
@@ -150,20 +144,37 @@ $ cd $GOPATH/src/github.com/openebs/maya
...
 
-#### After making changes to your local branch and running the `test`, create a pull request:
+While you rebase your changes, you will be required to resolve any conflicts that might arise and build and test your changes using the above steps. 
+
+## Submission
+
+### Create a pull request
+
+Before you raise the Pull Requests, make sure you have reviewed the checklist in the [CONTRIBUTING GUIDE](../CONTRIBUTING.md):
+- Ensure that you have re-based your changes with the upstream using the steps above. i
```
Comment (mahebbar) [inline docs/developer.md]: above. (delete i)

File: docs/developer.md [hunk 26]
```diff
@@ -150,20 +144,37 @@ $ cd $GOPATH/src/github.com/openebs/maya
...
+[Rebasing](https://git-scm.com/docs/git-rebase) is very import to keep your branch in sync with the changes being made by others and to avoid huge merge conflicts while raising your Pull Requests. You will always have to rebase before raising the PR. 
 
 ```sh
 # While on your myfeature branch (see above)
 git fetch upstream
 git rebase upstream/master
 ```
 
-#### After making changes to your local branch and running the `test`, create a pull request:
+While you rebase your changes, you will be required to resolve any conflicts that might arise and build and test your changes using the above steps.
```
Comment (mahebbar) [inline docs/developer.md]: you will be required to resolve = you must resolve

File: docs/developer.md [hunk 27]
```diff
@@ -150,20 +144,37 @@ $ cd $GOPATH/src/github.com/openebs/maya
...
 ```
 
-#### After making changes to your local branch and running the `test`, create a pull request:
+While you rebase your changes, you will be required to resolve any conflicts that might arise and build and test your changes using the above steps. 
+
+## Submission
+
+### Create a pull request
+
+Before you raise the Pull Requests, make sure you have reviewed the checklist in the [CONTRIBUTING GUIDE](../CONTRIBUTING.md):
```
Comment (mahebbar) [inline docs/developer.md]: make sure = ensure

File: CONTRIBUTING.md (hunk 2 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 3 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 5 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 6 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 7 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 8 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 9 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 10 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 11 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 12 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 13 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: README.md (hunk 14 above)
Comment (kmova) [reply README.md]: done

File: README.md (hunk 15 above)
Comment (kmova) [reply README.md]: done

File: README.md (hunk 16 above)
Comment (kmova) [reply README.md]: done

File: README.md (hunk 17 above)
Comment (kmova) [reply README.md]: done

File: README.md (hunk 18 above)
Comment (kmova) [reply README.md]: done

File: README.md (hunk 19 above)
Comment (kmova) [reply README.md]: done

File: README.md (hunk 21 above)
Comment (kmova) [reply README.md]: done

File: docs/developer.md (hunk 22 above)
Comment (kmova) [reply docs/developer.md]: done

File: docs/developer.md (hunk 23 above)
Comment (kmova) [reply docs/developer.md]: done

File: docs/developer.md (hunk 24 above)
Comment (kmova) [reply docs/developer.md]: it is more commands so leaving as procedure.

File: docs/developer.md (hunk 25 above)
Comment (kmova) [reply docs/developer.md]: done

File: docs/developer.md (hunk 26 above)
Comment (kmova) [reply docs/developer.md]: done

File: docs/developer.md (hunk 27 above)
Comment (kmova) [reply docs/developer.md]: done

**Comment by kmova** [comment]
Thanks @mahebbar for detailed comments. I have incorporated them and squashed with my previous commit.

File: CONTRIBUTING.md (hunk 3 above)
Comment (muratkars) [reply CONTRIBUTING.md]: first time - first-time

File: CONTRIBUTING.md [hunk 28]
```diff
@@ -1,46 +1,56 @@
...
 
-### Sign your work
-
-We use the Developer Certificate of Origin (DCO) as a additional safeguard
-for the OpenEBS project. This is a well established and widely used
-mechanism to assure contributors have confirmed their right to license
-their contribution under the project's license.
-Please read [dcofile](https://github.com/openebs/openebs/blob/master/contribute/developer-certificate-of-origin).
-If you can certify it, then just add a line to every git commit message:
+We use the Developer Certificate of Origin (DCO) as a additional safeguard for the OpenEBS project. This is a well established and widely used mechanism to assure that contributors have confirmed their right to license their contribution under the project's license. Please read [dcofile](https://github.com/openebs/openebs/blob/master/contribute/developer-certificate-of-origin). If you can certify it, then just add a line to every git commit message:
```
Comment (muratkars) [inline CONTRIBUTING.md]: a additional = an additional

File: CONTRIBUTING.md [hunk 29]
```diff
@@ -1,46 +1,56 @@
...
+# Contributing to OpenEBS Maya
 
-Are you ready to enhance Maya. Great!!!
+OpenEBS uses the standard GitHub pull requests process to review and accept contributions.  There are several areas that could use your help. For starters, you could help in improving the sections in this document by either creating a new issue describing the improvement or submitting a pull request to this repository. The issues for the various openebs components (including maya components) are maintained in [openebs/openebs](https://github.com/openebs/openebs/issues) repository.
 
-Feel free to contribute by:
-- raising issues
-- participate in commenting/voting on the issues
-- create new PRs for Documentation or Code
+* If you are a first time contributor, please see [Steps to Contribute](#steps-to-contribute).
```
Comment (muratkars) [inline CONTRIBUTING.md]: first time = first-time

File: CONTRIBUTING.md [hunk 30]
```diff
@@ -1,46 +1,56 @@
...
+You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases) like `git config --global alias.ci 'commit -s'`. Now you can commit with `git ci` and the commit will be signed.
+
+## Setting up your Development Environment
+
+This project is implemented using Go and uses the standard golang tools for development and build. In addition, this project heavily relies on Docker and Kubernetes. It is expected that the contributors:
+- are familiar with working with Go
+- are familiar with Docker containers
+- are familiar with Kubernetes and have access to a Kubernetes cluster or Minikube to test the changes.
+
+For setting up Development environment on your local host, see the detailed instructions [here](./docs/developer.md).
```
Comment (muratkars) [inline CONTRIBUTING.md]: For setting up Development environment = For setting up a Development environment

File: CONTRIBUTING.md [hunk 31]
```diff
@@ -1,46 +1,56 @@
-# Contributing to OpenEBS / Maya
+# Contributing to OpenEBS Maya
 
-Are you ready to enhance Maya. Great!!!
+OpenEBS uses the standard GitHub pull requests process to review and accept contributions.  There are several areas that could use your help. For starters, you could help in improving the sections in this document by either creating a new issue describing the improvement or submitting a pull request to this repository. The issues for the various openebs components (including maya components) are maintained in [openebs/openebs](https://github.com/openebs/openebs/issues) repository.
```
Comment (muratkars) [inline CONTRIBUTING.md]: openebs components = OpenEBS components

File: CONTRIBUTING.md (hunk 28 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 29 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: README.md [hunk 32]
```diff
@@ -3,34 +3,51 @@
...
+- Maintaining the inventory of the underlying disks on the Kubernetes Nodes 
+- Managing the allocation of Disks to CAS Storage Engines
+- Provisioning of CAS Volumes by interfacing with K8s Dynamic Volume Provisioner
+- Managing the high availability of the CAS volumes by tuning the scheduling parameters of CAS Deployments (Pods)
+- Provide adapters to CAS Volumes to interact with Kubernetes and related infrastructure components like Prometheus, Grafana etc.
 
-OpenEBS Maya allows you to manage storage across multiple zones (aka clusters/environments), that are co-located or geographically separated and can also run from within a single host. Maya can move the storage across different tiers based on the application needs (volume migration). Maya can learn and adapt to the changing environment through machine learning and data analytics. 
+*Maya* orchestration and management capabilities are delivered through a set of services and tools. Currently these services support deploying the CAS Storage Solutions in Kubernetes Clusters. In future, these can be extended to support other Container Orchestrators.
```
Comment (muratkars) [inline README.md]: Currently these = Currently, these (Missing comma after)

File: CONTRIBUTING.md (hunk 31 above)
Comment (kmova) [reply CONTRIBUTING.md]: done

File: docs/developer.md [hunk 33]
```diff
@@ -1,15 +1,18 @@
...
-### 1. Fork in the cloud
+## Initial Setup
+
+### Fork in the cloud
 
 1. Visit https://github.com/openebs/maya
 2. Click `Fork` button (top right) to establish a cloud-based fork.
 
-### 2. Clone fork to local storage
+### Clone fork to local machine
```
Comment (muratkars) [inline docs/developer.md]: local machine = local host (machine, host, node are all fine, but better to use same term, local host is used in other parts of the doc)

File: docs/developer.md [hunk 34]
```diff
@@ -1,15 +1,18 @@
-# Contributing Guide
+# Development Workflow
 
-To contribute openebs/maya firstly, you have to fork the openebs/maya repository. But before that you need to create an issue or pick an existing issue from [open issues](https://github.com/openebs/openebs/issues)
+## Prerequisites
 
-In any case, before you start working on your issue, sync your repository with the upstream openebs/maya master. Syncing ensures your repository has the latest changes.
+* You have Go 1.9 installed on your host/development machine.
+* You have Docker installed on your host/development machine. Docker is required for building maya component container images and to push them into Kubernetes Cluster for testing.
```
Comment (muratkars) [inline docs/developer.md]: into Kubernetes Cluster = into a Kubernetes Custer

File: README.md (hunk 32 above)
Comment (kmova) [reply README.md]: done

File: docs/developer.md (hunk 33 above)
Comment (kmova) [reply docs/developer.md]: done

File: docs/developer.md (hunk 34 above)
Comment (kmova) [reply docs/developer.md]: done
